packages/kbn-securitysolution-utils/src/path_validations/index.ts

```diff
@@ -0,0 +1,178 @@
+ */
+export const WIN_EXEC_PATH = /\\(\w+|\w*[\w+|-]+\/ +)+\w+[\w+|-]+\.*\w+$/i;
+export const UNIX_EXEC_PATH = /(\/|\w*[\w+|-]+\\ +)+\w+[\w+|-]+\.*\w*$/i;
+
+export const validateFilePathInput = ({
+  os,
+  value = '',
+}: {
+  os: OperatingSystem;
+  value?: string;
+}): string | undefined => {
+  const textInput = value.trim();
+  const isValidFilePath = isPathValid({
+    os,
+    field: 'file.path.text',
+    type: 'wildcard',
+    value: textInput,
+  });
+  const hasSimpleFileName = hasSimpleExecutableName({
+    os,
+    type: 'wildcard',
+    value: textInput,
+  });
+
+  if (!textInput.length) {
+    return FILEPATH_WARNING;
+  }
+
+  if (isValidFilePath) {
+    if (!hasSimpleFileName) {
+      return FILENAME_WILDCARD_WARNING;
+    }
+  } else {
+    return FILEPATH_WARNING;
+  }
+};
+
+export const hasSimpleExecutableName = ({
+  os,
+  type,
+  value,
+}: {
+  os: OperatingSystem;
+  type: TrustedAppEntryTypes;
+  value: string;
+}): boolean => {
+  if (type === 'wildcard') {
+    return os === OperatingSystem.WINDOWS ? WIN_EXEC_PATH.test(value) : UNIX_EXEC_PATH.test(value);
+  }
+  return true;
+};
+
+export const isPathValid = ({
+  os,
+  field,
+  type,
+  value,
+}: {
+  os: OperatingSystem;
+  field: ConditionEntryField | 'file.path.text';
+  type: TrustedAppEntryTypes;
+  value: string;
+}): boolean => {
+  if (field === ConditionEntryField.PATH || field === 'file.path.text') {
+    if (type === 'wildcard') {
+      return os === OperatingSystem.WINDOWS
+        ? isWindowsWildcardPathValid(value)
+        : isLinuxMacWildcardPathValid(value);
+    }
+    return doesPathMatchRegex({ value, os });
+  }
+  return true;
+};
+
+const doesPathMatchRegex = ({ os, value }: { os: OperatingSystem; value: string }): boolean => {
+  if (os === OperatingSystem.WINDOWS) {
+    const filePathRegex =
+      /^[a-z]:(?:|\\\\[^<>:"'/\\|?*]+\\[^<>:"'/\\|?*]+|%\w+%|)[\\](?:[^<>:"'/\\|?*]+[\\/])*([^<>:"'/\\|?*])+$/i;
+    return filePathRegex.test(value);
+  }
+  return /^(\/|(\/[\w\-]+)+|\/[\w\-]+\.[\w]+|(\/[\w-]+)+\/[\w\-]+\.[\w]+)$/i.test(value);
+};
+
+const isWindowsWildcardPathValid = (path: string): boolean => {
+  const firstCharacter = path[0];
+  const lastCharacter = path.slice(-1);
+  const trimmedValue = path.trim();
+  const hasSlash = /\//.test(trimmedValue);
+  if (path.length === 0) {
+    return false;
+  } else if (
+    hasSlash ||
+    trimmedValue.length !== path.length ||
+    firstCharacter === '^' ||
+    lastCharacter === '\\' ||
+    !hasWildcard({ path, isWindowsPath: true })
+  ) {
+    return false;
+  } else {
+    return true;
+  }
+};
+
+const isLinuxMacWildcardPathValid = (path: string): boolean => {
+  const firstCharacter = path[0];
+  const lastCharacter = path.slice(-1);
+  const trimmedValue = path.trim();
+  if (path.length === 0) {
+    return false;
+  } else if (
+    trimmedValue.length !== path.length ||
+    firstCharacter !== '/' ||
+    lastCharacter === '/' ||
+    path.length > 1024 === true ||
```

Comment:
nit: I'd add a comment/document this function about things such as the 1024 here. Is that a mapping thing?

Reply:
@paul-tavares I can't recall why we do this check. Would you happen to recall why we have this condition? I vaguely recall we lifted this from a legacy repo when we added the matches operator for trusted apps. 🤔